src/trace.rs

```diff
@@ -0,0 +1,136 @@
+use crate::event::command::{
+    CommandEventHandler,
+    CommandFailedEvent,
+    CommandStartedEvent,
+    CommandSucceededEvent,
+};
+use bson::Bson;
+
+pub(crate) struct CommandTracingEventEmitter {
+    max_document_length_bytes: usize,
+}
+
+impl CommandTracingEventEmitter {
+    pub(crate) fn new(max_document_length_bytes: Option<usize>) -> CommandTracingEventEmitter {
+        CommandTracingEventEmitter {
+            max_document_length_bytes: max_document_length_bytes.unwrap_or(1000),
+        }
+    }
+
+    fn serialize_command_or_reply(&self, doc: bson::Document) -> String {
+        let mut ext_json = doc.tracing_representation();
+        truncate_on_char_boundary(&mut ext_json, self.max_document_length_bytes);
+        ext_json
+    }
+}
+
+impl CommandEventHandler for CommandTracingEventEmitter {
+    fn handle_command_started_event(&self, event: CommandStartedEvent) {
```

Comment:
It seems like it might be worth recording command started -> command succeeded/failed as a span?

Reply:
I agree defining some spans of our own is a good idea, though for the purpose of this PR and the initial spec implementation I think I'd like to hold off on introducing the additional complexity of defining and testing spans until everything else is in place. it seems like something to revisit shortly after this work is finished though, and perhaps even before we initially release it. I've filed RUST-1390 about this.